### PR TITLE
Document/Elements/Paste-Style: Fix - When paste styling with globals,…

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: assets/dev/js/editor/document/elements/commands/paste-style.js [hunk 1]
```diff
@@ -96,26 +96,44 @@ export class PasteStyle extends CommandHistory {
...
-				targetContainer.view.allowRender = false;
+				this.pasteStyle( targetContainer, diffSettings );
+			} );
+		} );
+	}
 
-				// BC: Deprecated since 2.8.0 - use `$e.hooks`.
-				elementor.channels.data.trigger( 'element:before:paste:style', targetContainer.model );
+	pasteStyle( targetContainer, settings ) {
+		// Moved from `editor/elements/views/base.js` `pasteStyle` function.
```
Comment (matipojo) [inline assets/dev/js/editor/document/elements/commands/paste-style.js:105]: Remove comment.

File: assets/dev/js/editor/document/elements/commands/paste-style.js [hunk 2]
```diff
@@ -96,26 +96,44 @@ export class PasteStyle extends CommandHistory {
...
+		targetContainer.view.allowRender = false;
 
-				$e.run( 'document/elements/settings', {
-					container: targetContainer,
-					settings: diffSettings,
-					options: { external: true },
-				} );
+		// BC: Deprecated since 2.8.0 - use `$e.hooks`.
+		elementor.channels.data.trigger( 'element:before:paste:style', targetContainer.model );
```
Comment (matipojo) [inline assets/dev/js/editor/document/elements/commands/paste-style.js:109]: Add a deprecatio notice.

File: assets/dev/js/editor/document/elements/commands/paste-style.js [hunk 3]
```diff
@@ -96,26 +96,44 @@ export class PasteStyle extends CommandHistory {
...
+		if ( globals ) {
+			$e.run( 'document/globals/settings', {
+				container: targetContainer,
+				settings: globals,
+				render: false, // Prevent double render.
+			} );
+		}
+
+		// BC: Deprecated since 2.8.0 - use `$e.hooks`.
+		elementor.channels.data.trigger( 'element:after:paste:style', targetContainer.model );
```
Comment (matipojo) [inline assets/dev/js/editor/document/elements/commands/paste-style.js:132]: Add a deprecatio notice.

File: assets/dev/js/editor/document/elements/commands/paste-style.js [hunk 4]
```diff
@@ -96,26 +96,44 @@ export class PasteStyle extends CommandHistory {
...
+			container: targetContainer,
+			settings: settings,
+			options: { external: true },
 		} );
+
+		if ( globals ) {
+			$e.run( 'document/globals/settings', {
+				container: targetContainer,
+				settings: globals,
+				render: false, // Prevent double render.
```
Comment (matipojo) [inline assets/dev/js/editor/document/elements/commands/paste-style.js:127]: `options: { external: true },`?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/document/elements/commands/paste-style.js (hunk 4 above)
Comment (iNewLegend) [reply assets/dev/js/editor/document/elements/commands/paste-style.js:127]: options: { internal: true },?

File: assets/dev/js/editor/document/elements/commands/paste-style.js (hunk 2 above)
Comment (iNewLegend) [reply assets/dev/js/editor/document/elements/commands/paste-style.js:109]: https://github.com/elementor/elementor/blob/develop/assets/dev/js/editor/document/backwards-compatibility.js#L22

File: assets/dev/js/editor/document/elements/commands/paste-style.js (hunk 3 above)
Comment (iNewLegend) [reply assets/dev/js/editor/document/elements/commands/paste-style.js:132]: https://github.com/elementor/elementor/blob/develop/assets/dev/js/editor/document/backwards-compatibility.js#L22

File: assets/dev/js/editor/document/elements/commands/paste-style.js (hunk 1 above)
Comment (iNewLegend) [reply assets/dev/js/editor/document/elements/commands/paste-style.js]: Done

File: assets/dev/js/editor/document/elements/commands/paste-style.js (hunk 4 above)
Comment (iNewLegend) [reply assets/dev/js/editor/document/elements/commands/paste-style.js]: done

**Review by matipojo — CHANGES_REQUESTED** [review]
Tests?

**Comment by iNewLegend** [comment]
> Tests?

handled.

**Review by matipojo — APPROVED** [review]

**Review by kobizz — APPROVED** [review]

**Comment by rodolphebertozzo** [comment]
@KingYes

**Review by KingYes — APPROVED** [review]

**Pull request merged**